### PR TITLE
[RB] Try to set default branch env var for better snapshot matching

### PR DESCRIPTION
**Pull request by maggie-lou**
The snapshot key is based off the git branch. If we do not have a snapshot for a branch, we support falling back to snapshots for default branches, as specified in the `GIT_REPO_DEFAULT_BRANCH` and `GIT_BASE_BRANCH` env vars.
This change will enable more snapshot hits.

**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/hostedrunner/hostedrunner.go [hunk 1]
```diff
@@ -197,6 +188,13 @@ func (r *runnerService) createAction(ctx context.Context, req *rnpb.RunRequest,
 		})
 	}
 
+	for k, v := range req.GetEnv() {
```
Comment (maggie-lou) [inline enterprise/server/hostedrunner/hostedrunner.go:191]: When generating fallback snapshot keys, we only look at cmd.EnvironmentVariables, and not at the --env_overrides flag

File: .github/workflows/remote-bazel-build.yaml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
 
       - name: Test
         run: |
-          bb remote test //... \
+          bb remote --env=GIT_REPO_DEFAULT_BRANCH=master test //... \
```
Comment (bduffany) [inline .github/workflows/remote-bazel-build.yaml:25]: revert before merging?

File: enterprise/server/hostedrunner/hostedrunner.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/hostedrunner/hostedrunner.go:191]: Makes sense - do we need `env_overrides` anymore after this change or can we remove it?

**Review by bduffany — APPROVED** [review]

File: enterprise/server/hostedrunner/hostedrunner.go (hunk 1 above)
Comment (maggie-lou) [reply enterprise/server/hostedrunner/hostedrunner.go:191]: No we don't. Cleaned up in workflow service too. Don't know why I didn't implement this way to begin with...

File: .github/workflows/remote-bazel-build.yaml (hunk 2 above)
Comment (maggie-lou) [reply .github/workflows/remote-bazel-build.yaml:25]: The remote bazel code determines the default branch from the local git state. The github runner only checks out one commit, so we'd have to do that hacky thing where we briefly checkout master in order to take advantage of that. It's easier to just explicitly pass the flag

**Pull request merged**